src/models/discomfort_index.js

```diff
@@ -0,0 +1,98 @@
+import { round } from "../utilities/utilities";
+
+/**
+ * @typedef {Object} DiscomfortIndexReturnType - a result set containing the discomfort index and the classification of the
+ * thermal comfort conditions
+ * @property {number} di – Discomfort Index(DI)
+ * @property {string} discomfort_condition Classification of the thermal comfort conditions according to the discomfort index
+ */
+
+/**
+ * @public
+ * Calculates the Discomfort Index (DI). The index is essentially an effective temperature based on air temperature and humidity.
+ * The discomfort index is usuallly divided in 6 dicomfort categories and it only applies to warm environments.
+ * @see {@link discomfort_index_array} for a version that supports arrays
+ *
+ * @param {number} tdb - air temperature [C]
+ * @param {number} rh - relative humidity [%]
+ *
+ * @returns {DiscomfortIndexReturnType} object with results of DI
+ * @example
+ * const DI = discomfort_index(25, 50); // returns { di: 22.1, discomfort_condition: 'Less than 50% feels discomfort' }
+ */
+export function discomfort_index(tdb, rh) {
+  const di = calculate_di(tdb, rh);
+  const condition = check_categories(di);
+
+  return {
+    di: round(di, 1),
+    discomfort_condition: condition,
+  };
+}
+
+/**
+ * @typedef {Object} DiscomfortIndexArrayReturnType - a result set containing an array of discomfort index and an array
+ * of classification of the thermal comfort conditions
+ * @property {number[]} di – Discomfort Index(DI) Array
+ * @property {string[]} discomfort_condition Classification of the thermal comfort conditions in array
+ */
+
+/**
+ * @public
+ * Calculates the Discomfort Index (DI). The index is essentially an effective temperature based on air temperature and humidity.
+ * The discomfort index is usuallly divided in 6 dicomfort categories and it only applies to warm environments.
+ * @see {@link discomfort_index} for a version that supports scalar arguments
+ *
+ * @param {number[]} tdb - air temperature [C]
+ * @param {number[]} rh - relative humidity [%]
+ *
+ * @returns {DiscomfortIndexArrayReturnType} object with results of DI
+ */
+export function discomfort_index_array(tdb, rh) {
+  const di = tdb.map((temperature, index) =>
+    calculate_di(temperature, rh[index]),
+  );
+
+  const discomfortCondition = di.map((value) => check_categories(value));
+
+  return {
+    di: di.map((value) => Math.round(value * 10) / 10),
```

Comment:
use round from utilities, like you did above

Reply:
done!